examples/thread_test/Cargo.toml

```
@@ -0,0 +1,7 @@
+[package]
+name = "thread_test"
+authors = ["Marco Ruppert-Maas <marco.ruppert-maas@studium.fernuni-hagen.de>"]
+edition = "2021"
```

Can we make this Rust 2024?

```suggestion
edition = "2024"
```